File: ckgame-java/src/main/java/com/ckgame/politics/FactionEvent.java
package com.ckgame.politics;

import com.ckgame.core.Constants;

import java.util.ArrayList;
import java.util.List;

/**
 * 派系相关事件。
 */
public final class FactionEvent {
    public String kind;
    public int factionId;
    public int liege = 0;
    public int founder = 0;
    public int who = 0;
    public FactionKind factionKind = null;
    public final List<Integer> members = new ArrayList<>();
    public String reason = "";

    public FactionEvent(String kind, int factionId) {
        this.kind = kind;
        this.factionId = factionId;
    }

    public FactionEvent(String kind, int factionId, int liege, int founder, int who,
                        FactionKind factionKind, List<Integer> members, String reason) {
        this.kind = kind;
        this.factionId = factionId;
        this.liege = liege;
        this.founder = founder;
        this.who = who;
        this.factionKind = factionKind;
        if (members != null) {
            this.members.addAll(members);
        }
        this.reason = reason != null ? reason : "";
    }
}

File: ckgame-ts/src/politics/FactionEvent.ts
import { FactionKind } from './FactionKind.js';

/**
 * Faction-related event.
 */
export interface FactionEvent {
  kind: string;
  factionId: number;
  liege: number;
  founder: number;
  who: number;
  factionKind?: FactionKind;
  members?: number[];
  reason?: string;
}

/**
 * Create a FactionEvent with defaults.
 */
export function createFactionEvent(
  kind: string,
  factionId: number,
  liege: number = 0,
  founder: number = 0,
  who: number = 0,
  factionKind?: FactionKind,
  members?: number[],
  reason?: string,
): FactionEvent {
  return {
    kind,
    factionId,
    liege,
    founder,
    who,
    factionKind,
    members: members ? [...members] : undefined,
    reason: reason ?? '',
  };
}
